scripts/DllManipulatorScript.cs

```diff
@@ -84,12 +87,29 @@ private void Initialize()
         {
             var initTimer = System.Diagnostics.Stopwatch.StartNew();
 
-            DllManipulator.Options = Options;
-            DllManipulator.Initialize(Thread.CurrentThread.ManagedThreadId, Application.dataPath);
+            DllManipulator.Initialize(Options, Thread.CurrentThread.ManagedThreadId, Application.dataPath);
 
             initTimer.Stop();
             InitializationTime = initTimer.Elapsed;
         }
+
+        /// <summary>
+        /// Will reset the DllManipulator and Initialize it again.
+        /// Note: Unloads all Dlls, may be a dangerous operation if using preloaded
+        /// </summary>
+        public void Reinitialize()
+        {
+            if(_singletonInstance != this)
+                return;
+            
+            if (DllManipulator.Options != null)
+                DllManipulator.Reset();
+            
+#if UNITY_EDITOR
```

Comment:
I think if this is not going to actually perform the initialization again, then this method shouldn't be called in the first place (it doesn't actually reinitialize).

Reply:
Yes, I've changed it although this requires that the Initialize function is public.